tutorials/getstarted.qmd

```diff
@@ -0,0 +1,471 @@
+"""
+
+```{julia}
+function generic_de_Casteljau(
+    M::AbstractManifold,
+    t,
+    pts::NTuple{2};
+    m::AbstractRetractionMethod=default_retraction_method(M),
+    n::AbstractInverseRetractionMethod=default_inverse_retraction_method(M),
+)
+    X = inverse_retract(M, pts[1], pts[2], n)
+    return retract(M, pts[1], X, t, m)
+end
+```
+
+and for the recursion
+
+```{julia}
+function generic_de_Casteljau(
+    M::AbstractManifold,
+    t,
+    pts::NTuple;
+    m::AbstractRetractionMethod=default_retraction_method(M),
+    n::AbstractInverseRetractionMethod=default_inverse_retraction_method(M),
+)
+    p = generic_de_Casteljau(M, t, pts[1:(end - 1)]; m=m, n=n)
+    q = generic_de_Casteljau(M, t, pts[2:end]; m=m, n=n)
+    X = inverse_retract(M, p, q, n)
+    return retract(M, p, X, t, m)
+end
+```
+
+Note that on a manifold `M` where the exponential map is implemented, the `default_retraction_method(M)` returns [🔗 `ExponentialRetraction`](https://juliamanifolds.github.io/ManifoldsBase.jl/stable/retractions.html#ManifoldsBase.ExponentialRetraction), which yields that the `retract` function falls back to calling `exp`.
+
+The same mechanism exists for [🔗 `parallel_transport_to(M, p, X, q)`](https://juliamanifolds.github.io/ManifoldsBase.jl/stable/functions.html#ManifoldsBase.parallel_transport_to-Tuple{AbstractManifold,%20Any,%20Any,%20Any}) and the more general [🔗 `vector_transport_to(M, p, X, q, m)`](https://juliamanifolds.github.io/ManifoldsBase.jl/stable/vector_transports.html#ManifoldsBase.vector_transport_to) whose [🔗 `AbstractVectorTransportMethod`](https://juliamanifolds.github.io/ManifoldsBase.jl/stable/vector_transports.html#ManifoldsBase.AbstractVectorTransportMethod) `m` has a default defined by [🔗 `default_vector_transport_method(M)`](https://juliamanifolds.github.io/ManifoldsBase.jl/stable/vector_transports.html#ManifoldsBase.default_vector_transport_method-Tuple{AbstractManifold}).
+
+## Allocating and in-place computations
+
+Memory allocation is a [critical performace issue](https://docs.julialang.org/en/v1/manual/performance-tips/#Measure-performance-with-[@time](@ref)-and-pay-attention-to-memory-allocation) when programming in Julia. To take this into account, `Manifolds.jl` provides special functions to reduce the amount of allocations.
+
+We again look at the [📖 exponential map](https://en.wikip edia.org/wiki/Exponential_map_(Riemannian_geometry)). On a manifold `M` the exponential map needs a point `p` (to start from) and a tangent vector `X`, which can be seen as direction to “walk into” as well as the length to walk into this direction. In `Manifolds.jl` the function can then be called with `q = exp(M, p, X)` [🔗](https://juliamanifolds.github.io/ManifoldsBase.jl/stable/functions.html#Base.exp-Tuple{AbstractManifold,%20Any,%20Any}). This function returns the resulting point `q`, which requires to allocate new memory.
```

Comment:
Links where only the chain icon is clickable look inconsistent with the rest of this document.

Reply:
And some links don't have the chain icon. Is that intentional? For example below the link to `BuresWassersteinMetric`.

Reply:
the chain icon meant (in the Pluto still) to refer to links outside the Manifolds.jl docs – here one could link the whole `exp` but since then we have to put the chain behind, that would still be inconsistent.